WebARonARKit/WebARonARKit/ViewController.m

```diff
@@ -952,40 +998,66 @@ - (void)session:(ARSession *)session didUpdateFrame:(ARFrame *)frame {
-                        pProjectionMatrix[15]];
-
+    // Get the camera frame in base 64.
+    NSString* base64ImageString = @"";
+    if (USE_CAMERA_FRAME) {
+      base64ImageString = [self getBase64ImageFromPixelBuffer:
+                           frame.capturedImage];
+      if (!base64ImageString) {
+        base64ImageString = @"";
+      }
+      else {
+        base64ImageString = [NSString stringWithFormat:
+                             @"data:image/jpg;base64, %@", base64ImageString];
+      }
+    }
+  
+    // Create a NSDictionary that will be parsed as a json and then passed to the JS side
+    NSDictionary* jsonDictionary = @{
+        @"position":@[FBOX(position[0]), FBOX(position[1]), FBOX(position[2])],
+        @"orientation":@[FBOX(pOrientationQuat[0]), FBOX(pOrientationQuat[1]),
+                         FBOX(pOrientationQuat[2]), FBOX(pOrientationQuat[3])],
+        @"viewMatrix":@[FBOX(pViewMatrix[0]), FBOX(pViewMatrix[1]),
+                        FBOX(pViewMatrix[2]), FBOX(pViewMatrix[3]),
+                        FBOX(pViewMatrix[4]), FBOX(pViewMatrix[5]),
+                        FBOX(pViewMatrix[6]), FBOX(pViewMatrix[7]),
+                        FBOX(pViewMatrix[8]), FBOX(pViewMatrix[9]),
+                        FBOX(pViewMatrix[10]), FBOX(pViewMatrix[11]),
+                        FBOX(pViewMatrix[12]), FBOX(pViewMatrix[13]),
+                        FBOX(pViewMatrix[14]), FBOX(pViewMatrix[15])],
+        @"projectionMatrix":@[FBOX(pProjectionMatrix[0]),
+                              FBOX(pProjectionMatrix[1]),
+                              FBOX(pProjectionMatrix[2]),
+                              FBOX(pProjectionMatrix[3]),
+                              FBOX(pProjectionMatrix[4]),
+                              FBOX(pProjectionMatrix[5]),
+                              FBOX(pProjectionMatrix[6]),
+                              FBOX(pProjectionMatrix[7]),
+                              FBOX(pProjectionMatrix[8]),
+                              FBOX(pProjectionMatrix[9]),
+                              FBOX(pProjectionMatrix[10]),
+                              FBOX(pProjectionMatrix[11]),
+                              FBOX(pProjectionMatrix[12]),
+                              FBOX(pProjectionMatrix[13]),
+                              FBOX(pProjectionMatrix[14]),
+                              FBOX(pProjectionMatrix[15])]
+        ,@"cameraFrame":base64ImageString
+    };
+    // Pass the dictionary to JSON and back to a string.
+    NSError* error;
+    NSData* jsonData = [NSJSONSerialization dataWithJSONObject:jsonDictionary options:NSJSONWritingPrettyPrinted error:&error];
+    NSString* jsonString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+  
+//    NSLog(@"jsonString = %@", jsonString);
+
+    // This will be the final JS code to evaluate
+    NSString* jsCode =
+        [NSString stringWithFormat:
+        @"if (window.WebARonARKitSetData) "
```

Comment:
continuation indent

Reply:
But, if the previous line was already a continuation indent, do I have to apply a new continuation indent? I think I do not fully understand this yet :(

Reply:
maybe I am confused about objective-c syntax, but I believe the [NSString stringWithFormat:.... is a new function call. Continuation indents nest if you open new scopes. To use a simpler language as an example:

float thisVariableNameIsSoLongYouCantFitTheFunctionNameOnTheSameLine =
    thisFunctionNameIsAlsoSoLongYouCantFitTheParametersOnTheSameLine(
        thisParameterIsNowAtTheSecondLevelContinuationIndent);

Does that make sense? The third line is in a new scope so it gets a continuation indent off of the already-continuation-indented previous line. Hope that makes sense!